File: src/server/Services/MeetingService.cs
using AutoMapper;
using Microsoft.EntityFrameworkCore;
using WebRTCVideoCallAssistant.Server.Models;
using WebRTCVideoCallAssistant.Server.Models.Dto;

namespace WebRTCVideoCallAssistant.Server.Services;

public class MeetingService
{
    private readonly AppDbContext _db;
    private readonly IMapper _mapper;
	
	public MeetingService(AppDbContext db, IMapper mapper)
	{
		_db = db;
		_mapper = mapper;
	}

	public Meeting Create(CreateMeetingDto dto)
	{
        var meeting = _mapper.Map<Meeting>(dto);

		if (meeting.StartsAt < DateTime.Now)
			throw new ApplicationException("Meeting cannot be created for a past time");

		meeting.UserConnId = Guid.NewGuid().ToString();
		meeting.CustomerConnId = Guid.NewGuid().ToString();
		
		var len = 8;
		var buffer = new char[len];
		var random = new Random();
		var alph = "abcdefghijklmnopqrstuvwxyz";

		for (int i = 0; i < len; i++)
			buffer[i] = alph[random.Next(alph.Length)];
		meeting.Slug = string.Join("", buffer);
		
        var res = _db.Meetings.Add(meeting).Entity;
        _db.SaveChanges();

		return res;
	}

	public Meeting Get(int id)
	{
		return GetById(id);
	}

	public IEnumerable<Meeting> GetAll()
	{
		return _db.Meetings
			.Include(m => m.CreatedBy)
			.Include(m => m.CreatedFor)
			.Include(m => m.Stat);
	}

	public IQueryable<Meeting> GetAllByUser(int userId)
	{
		return _db.Meetings.Where(m => m.CreatedById == userId)
			.Include(m => m.CreatedBy)
			.Include(m => m.CreatedFor)
			.Include(m => m.Stat);
	}

	public Meeting Update(int id, UpdateMeetingDto dto)
    {
		if (dto?.StartsAt < DateTime.Now)
			throw new ApplicationException("Meeting cannot be created for a past time");

        var meeting = GetById(id);

        _mapper.Map(dto, meeting);
        var res = _db.Meetings.Update(meeting).Entity;
        _db.SaveChanges();

		return res;
    }

	public Meeting Delete(int id)
    {
        var meeting = GetById(id);

        var res = _db.Meetings.Remove(meeting).Entity;
        _db.SaveChanges();

		return res;
    }

	public Meeting ResolveSlug(string slug)
	{
		var meeting = _db.Meetings		
			.Include(m => m.CreatedBy)
			.Include(m => m.CreatedFor)
			.Include(m => m.Stat)
			.FirstOrDefault(i => i.Slug == slug);
		if (meeting == null)
			throw new KeyNotFoundException($"Meeting with slug '{slug}' not found");
		return meeting;
	}

	private Meeting GetById(int id)
	{
		var meeting = _db.Meetings
			.Include(m => m.CreatedBy)
			.Include(m => m.CreatedFor)
			.Include(m => m.Stat)
			.FirstOrDefault(m => m.Id == id);
        if (meeting == null) throw new KeyNotFoundException("Meeting not found");
        return meeting;
	}
}

File: src/client/src/app/services/meeting.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { Meeting, MeetingCreate } from '../models/meeting';
import { Stat } from "../models/stat";

@Injectable({
  providedIn: 'root'
})
export class MeetingService {

  constructor(private http: HttpClient) { }

  public get(id: number) {
    return this.http.get<Meeting>(`${environment.API_URL}Meeting/Get/${id}`);
  }

  public getAll() {
    return this.http.get<Meeting[]>(`${environment.API_URL}Meeting/GetAll`);
  }

  public getAllByUser(userId:number) {
    return this.http.get<Meeting[]>(`${environment.API_URL}Meeting/GetAllByUser/${userId}`);
  }

  public create(meeting: MeetingCreate) {
    return this.http.put<Meeting>(`${environment.API_URL}Meeting/Create`, meeting);
  }

  public update(meeting: Meeting) {
    return this.http.patch(`${environment.API_URL}Meeting/Update/${meeting.id}`, meeting);
  }

  public delete(id: number) {
    return this.http.delete<Meeting>(`${environment.API_URL}Meeting/Delete/${id}`);
  }

  public resolveSlug(slug: string) {
    return this.http.get<Meeting>(`${environment.API_URL}Meeting/ResolveSlug/${slug}`);
  }

  public createStat(stats: any) {
	return this.http.put<Stat>(`${environment.API_URL}Stat/Create`, stats);
  }
}
